src/daft-micropartition/src/ops/join.rs

```diff
@@ -41,6 +41,7 @@ impl MicroPartition {
             return Ok(Self::empty(Some(join_schema.into())));
         }
 
+        // TODO(Clark): Elide concatenations where possible by doing a chunk-aware local table join.
```

Comment:
Was this observed to be significant in terms of performance? Wonder if its worth making a ticket here

Reply:
I haven't profiled heavily enough yet to determine if this is a real bottleneck, but it does seem like a relatively easy win! The join algorithm ultimately consists of a straightforward iteration through both the build side and the probe side, so there's no reason that we couldn't iterate directly on the chunks rather than concatenating and then iterating.